spec/loaders/abstract-loader-shared-behavior.coffee

```diff
@@ -303,6 +303,26 @@ registerSharedBehavior "AbstractLoaderSharedBehavior", (sharedContext) ->
           loader.setup(opts)
           notFound(loader, opts)
 
+      context 'when optional fields have been requested but the fields arent on the tasks', ->
+        beforeEach ->
+          opts.optionalFields = ['test_field']
+          taskOne.set('test_field', 'fake value')
+          loader.storageManager.storage('tasks').add([taskOne, taskTwo])
+
+        it 'returns false and does not call #_onLoadSuccess', ->
+          loader.setup(opts)
+          loader._checkCacheForData()
+          expect(loader._onLoadSuccess).not.toHaveBeenCalled()
+
+      context 'when optional fields have been requested but the fields are there on the tasks', ->
+        beforeEach ->
+          loader.storageManager.storage('tasks').add([taskOne, taskTwo])
```

Comment:
Don't we need to be doing something like `taskOne.set('test_field', 'fake value')` for both tasks in this context?